### DISCUSSION TITLE
EOApi Best Practices

### DISCUSSION THREAD

**Discussion by julianblue** [developmentseed/eoAPI#147] [Q&A]
Hi, 

I looked through the demos and was wondering about the best practices and different approaches to adding stac items and collections to the db. Is the best practice to add a new stac collection to use pypgstac ? Would this also be the approach for adding single stac items or is there a way to do this in python ?

Thanks in advance.

**Comment by zacdezgeo** [comment]
Hey! Thanks for opening the discussion. When it comes to ingestion, we’re still discovering best practices. We use different approaches with slight variations based on the project, as much on the orchestration side as on the ingestion side. We are gathering feedback from others to inform our understanding of best practices.

Currently, the approach suggested in the demos can be used: creating a file with the stac items to be ingested and ingesting it with pypgstac. For inspiration, you can check out the details in this [example](https://github.com/vincentsarago/MAXAR_opendata_to_pgstac/blob/main/generate.py). We also just released a new version of [eoAPI-cdk](https://github.com/developmentseed/eoapi-cdk) if you are in the AWS ecosystem with an [STAC ingestor service](https://developmentseed.org/eoapi-cdk/#stacingestor-).

Can you explain how you would like to ingest stac items? Are there particular reasons for your use case of needing to ingest a single item?
